test/e2e/utils.go

```diff
@@ -213,6 +216,31 @@ func undeployJaegerInstance(jaeger *v1.Jaeger) {
 	require.NoError(t, err)
 }
 
+func getJaegerInstance(name, namespace string) *v1.Jaeger {
+	jaegerInstance := &v1.Jaeger{}
+	key := types.NamespacedName{Name: name, Namespace: namespace}
+	err := fw.Client.Get(goctx.Background(), key, jaegerInstance)
+	require.NoError(t, err)
+	return jaegerInstance
+}
+
+type ValidateHttpResponseFunc func(response *http.Response) (done bool, err error)
+
+func WaitAndPollForHttpResponse(targetUrl string, condition ValidateHttpResponseFunc) (err error) {
```

Comment:
If you aren't specifying anything else for this type, you can just use the `func` definition as you had there as the parameter to this func. Like:

```go
 func WaitAndPollForHttpResponse(targetUrl string, condition func(response *http.Response) (done bool, err error)) (err error) {
```

Reply:
On a second thought, I think your way is easier to read... Don't listen to me.

Reply:
:-)  I based it on this https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go#L202-L204 which is also not strictly necessary, but makes things a bit clearer.